src/crimson/os/seastore/journal/circular_bounded_journal.cc

```diff
@@ -117,17 +117,15 @@ RecordScanner::read_validate_record_metadata_ret CircularBoundedJournal::read_va
 {
   LOG_PREFIX(CircularBoundedJournal::read_validate_record_metadata);
   paddr_t start = cursor.seq.offset;
-  return read_record(start, 0
+  return read_record(start, nonce
   ).safe_then([FNAME, &cursor](auto ret) {
     if (!ret.has_value()) {
       return read_validate_record_metadata_ret(
 	read_validate_record_metadata_ertr::ready_future_marker{},
 	std::nullopt);
     }
     auto [r_header, bl] = *ret;
-    if ((cursor.last_committed != JOURNAL_SEQ_NULL &&
-	cursor.last_committed > r_header.committed_to) ||
-	r_header.committed_to.segment_seq != cursor.seq.segment_seq) {
+    if (r_header.committed_to.segment_seq != cursor.seq.segment_seq) {
       DEBUG("invalid header: {}", r_header);
```

Comment:
Apart from the above comment, can `read_validate_record_metadata` and `read_validate_data` be shared?

I don't see how the two implementations are special apart from:
1. getting the block size.
2. getting the segment size (for RBM, it is the end of the CBJ)
3. generalize read(paddr, len) -> bufferptr

If that's true, probably 1 and 2 can be passed to `cursor` or to `scan_valid_records`, and `RecordScanner` only needs a virtual `read()` interface. Does it make sense?